core/unit_test/TestGraph.hpp

```diff
@@ -739,4 +739,61 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), end_of_submit_control_flow) {
             value_A + 2 * value_B + value_C + value_D + value_E + value_F);
 }
 
+template <typename Exec>
+struct GraphNodeTypes {
+  //! Type of a kernel node built using a Kokkos parallel construct.
+  using kernel_t =
+      Kokkos::Impl::GraphNodeKernelImpl<Exec, Kokkos::RangePolicy<Exec>,
+                                        CountTestFunctor<Exec>,
+                                        Kokkos::ParallelForTag>;
+
+  //! Type of an aggregate node.
+  using aggregate_t = typename Kokkos::Impl::GraphImpl<Exec>::aggregate_impl_t;
+};
+
+TEST(TEST_CATEGORY, is_graph_kernel_v) {
```

Comment:
Not a good place for this comment, but: should there now be a concept `is_graph_aggragate_v` too? It could also be introduced later when a concrete use case should arise.

Reply:
Indeed, for now there wouldn't be any use case for such a concept, so let's not clutter this PR.